### DISCUSSION TITLE
Customizing Cockpit with Custom Apps...

### DISCUSSION THREAD

**Discussion by bstin** [cockpit-project/cockpit#20007] [General]
Hi. 

I am a long-time user of Cockpit. Its a truly great product that has made my sysadmin duties much easier! **Thanks for developing it!**

To my question: I would like to develop some custom apps / customization's for our internal team use. Having said that, *I am not a developer*! 

I've looked at the starter-project and it seems simple enough, but I notice its based around React. Unfortunately for me, this is bit of a stretch....I wouldn't be able to do much more than "hello world" in React....so this brings me to my question(s): Would any of these things be possible / easy to do?

1. Our company has an internal documentation site that is generated from `mkdocs` (if you don't know thats a static website generator...you feed it markdown files and it spits out a bare html static website). So, could I create a custom "doc app" that was little more than just pointing to this directory of static html / css files? 

2. A different example is we use Zabbix as our monitoring solution. Could I create an iframe and embed our Zabbix dashboard as an "app"? Maybe cross-origin might be an issue, but I could just put a proxy server on same cockpit machine and point the "app" to localhost:3000 (or whatever) and let that feed from the remote zabbix dashboard? Possible?

3. Lastly, personally I've hacked together some websites using html + jquery in the past (thats my limit of being a web developer). Is it necessary that one writes an app in React? Or could apps be written in more traditional html / javascript? The prospect of being whip up some html+js and tap into system tools and "webify" them is very interesting.

Thanks for your help and for all your teams great work.

**Comment by martinpitt** [comment]
1. It's certainly possible to create a little cockpit page that watches a given docs source directory, runs `mkdocs` on changes, and either opens the generated docs in an embedded iframe or just a separate browser tab.
2. Possible if they both come from the same origin (that's browser enforced), i.e. you run both zabbix and cockpit under the same reverse proxy.
3. You don't *have* to use React/PF for a Cockpit app. You can use bare-bones HTML/JS without any "build system" overhead, see https://cockpit-project.org/blog/creating-plugins-for-the-cockpit-user-interface.html . Of course this won't look pretty, but for some "quick personal hack" scenarios that is just right. There's also other apps written in Vue etc.; it's just our recommendation for "officially packaged" cockpit apps to maintain a consistent and accessible UI.
